test/skimNtuple_HHbtag.cpp

```diff
@@ -2172,118 +2184,118 @@ int main (int argc, char** argv)
-	  theSmallTree.m_MC_pdf77 = (isMC ? theBigTree.MC_pdf[77] : 1.);
-	  theSmallTree.m_MC_pdf78 = (isMC ? theBigTree.MC_pdf[78] : 1.);
-	  theSmallTree.m_MC_pdf79 = (isMC ? theBigTree.MC_pdf[79] : 1.);
-	  theSmallTree.m_MC_pdf80 = (isMC ? theBigTree.MC_pdf[80] : 1.);
-	  theSmallTree.m_MC_pdf81 = (isMC ? theBigTree.MC_pdf[81] : 1.);
-	  theSmallTree.m_MC_pdf82 = (isMC ? theBigTree.MC_pdf[82] : 1.);
-	  theSmallTree.m_MC_pdf83 = (isMC ? theBigTree.MC_pdf[83] : 1.);
-	  theSmallTree.m_MC_pdf84 = (isMC ? theBigTree.MC_pdf[84] : 1.);
-	  theSmallTree.m_MC_pdf85 = (isMC ? theBigTree.MC_pdf[85] : 1.);
-	  theSmallTree.m_MC_pdf86 = (isMC ? theBigTree.MC_pdf[86] : 1.);
-	  theSmallTree.m_MC_pdf87 = (isMC ? theBigTree.MC_pdf[87] : 1.);
-	  theSmallTree.m_MC_pdf88 = (isMC ? theBigTree.MC_pdf[88] : 1.);
-	  theSmallTree.m_MC_pdf89 = (isMC ? theBigTree.MC_pdf[89] : 1.);
-	  theSmallTree.m_MC_pdf90 = (isMC ? theBigTree.MC_pdf[90] : 1.);
-	  theSmallTree.m_MC_pdf91 = (isMC ? theBigTree.MC_pdf[91] : 1.);
-	  theSmallTree.m_MC_pdf92 = (isMC ? theBigTree.MC_pdf[92] : 1.);
-	  theSmallTree.m_MC_pdf93 = (isMC ? theBigTree.MC_pdf[93] : 1.);
-	  theSmallTree.m_MC_pdf94 = (isMC ? theBigTree.MC_pdf[94] : 1.);
-	  theSmallTree.m_MC_pdf95 = (isMC ? theBigTree.MC_pdf[95] : 1.);
-	  theSmallTree.m_MC_pdf96 = (isMC ? theBigTree.MC_pdf[96] : 1.);
-	  theSmallTree.m_MC_pdf97 = (isMC ? theBigTree.MC_pdf[97] : 1.);
-	  theSmallTree.m_MC_pdf98 = (isMC ? theBigTree.MC_pdf[98] : 1.);
-	  theSmallTree.m_MC_pdf99 = (isMC ? theBigTree.MC_pdf[99] : 1.);
-	  theSmallTree.m_MC_pdf100 = (isMC ? theBigTree.MC_pdf[100] : 1.);
- 
-	  theSmallTree.m_MC_astrong0 = (isMC ? theBigTree.MC_astrong[0] : 1.);
-	  theSmallTree.m_MC_astrong1 = (isMC ? theBigTree.MC_astrong[1] : 1.);
+	  theSmallTree.m_MC_QCDscale0 = (isMC ? (theBigTree.MC_QCDscale[0] != 0 ? theBigTree.MC_QCDscale[0] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+	  theSmallTree.m_MC_QCDscale1 = (isMC ? (theBigTree.MC_QCDscale[1] != 0 ? theBigTree.MC_QCDscale[1] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+	  theSmallTree.m_MC_QCDscale2 = (isMC ? (theBigTree.MC_QCDscale[2] != 0 ? theBigTree.MC_QCDscale[2] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+	  theSmallTree.m_MC_QCDscale3 = (isMC ? (theBigTree.MC_QCDscale[3] != 0 ? theBigTree.MC_QCDscale[3] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+	  theSmallTree.m_MC_QCDscale4 = (isMC ? (theBigTree.MC_QCDscale[4] != 0 ? theBigTree.MC_QCDscale[4] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+	  theSmallTree.m_MC_QCDscale5 = (isMC ? (theBigTree.MC_QCDscale[5] != 0 ? theBigTree.MC_QCDscale[5] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+	  theSmallTree.m_MC_QCDscale6 = (isMC ? (theBigTree.MC_QCDscale[6] != 0 ? theBigTree.MC_QCDscale[6] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
+
+	  theSmallTree.m_MC_pdf0 = (isMC ? (theBigTree.MC_pdf[0] != 0 ? theBigTree.MC_pdf[0] * MC_weight_fixed : theSmallTree.m_MC_weight) : 1.);
```

Comment:
what if the (here PDF) weight really is zero?

Reply:
At the skimming stage I don't think we can do anything about that, but I don't think this can happen to be honest